realm/realm-library/src/main/java/io/realm/internal/SharedGroup.java

```diff
@@ -293,6 +297,29 @@ public int hashCode() {
         }
     }
 
+    /**
+     * Waits for change committed by {@link SharedGroup} in other Thread.
+     *
+     * @return {@code true} if successfully detects change, {@code false} no change has detected otherwise.
+     */
+    public boolean waitForChange() {
+        if (isStoppedWaitingByUser) {
+            isStoppedWaitingByUser = false;
+            return false;
+        }
```

Comment:
I doubt the right way for user to use these pair of methods would be some thing like:

```java
while (waitingThread.isAlive()) {
    realm.stopWaitForChange();
}